_rules/menuitem-non-empty-name-m6b1q3.md

```diff
@@ -0,0 +1,161 @@
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - CSS Styling
+  - DOM Tree
+acknowledgments:
+  authors:
+    - Wilco Fiers
+  assets:
+    - The "file.svg" icon is part of the Feather Icons package, which uses the [MIT license](https://github.com/feathericons/feather/blob/master/LICENSE).
+---
+
+## Applicability
+
+This rule applies to HTML elements with a `menuitem` [semantic role][] that are [included in the accessibility tree][].
+
+## Expectation
+
+Each target element has an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+This rule assumes that all menuitems are [user interface components as defined by WCAG 2](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). If an element has a role of `menuitem` that would not be perceived as a single control by users, [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value) would not apply and so failing this rule would not result in a conformance issue.
+
+## Accessibility Support
+
+Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `button` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
+
+## Background
+
+- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)
+- [ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA14)
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA16)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This element with the `menuitem` role has an [accessible name][] because of its text content.
+
+```html
+<ul role="menu">
+	<li role="menuitem">New file</li>
+</ul>
+```
+
+#### Passed Example 2
+
+This element with the `menuitem` role has an [accessible name][] because of its `aria-label` attribute.
+
+```html
+<ul role="menu">
+	<li role="menuitem" aria-label="New file">
+		<img src="/test-assets/shared/file.svg" alt="" />
+	</li>
+</ul>
+```
+
+#### Passed Example 3
+
+This element with the `menuitem` role has an [accessible name][] because its `aria-labelledby` attribute references an element with text content.
+
+```html
+<ul role="menu">
+	<li role="menuitem" aria-labelledby="newfile">
+		<img src="/test-assets/shared/file.svg" alt="" />
+		<span id="newfile" hidden>New file</span>
+	</li>
+</ul>
+```
+
+#### Passed Example 4
+
+This element with the `menuitem` role has an [accessible name][] because of its `title` attribute.
+
+```html
+<ul role="menu">
+	<li role="menuitem" title="New file">
+		<img src="/test-assets/shared/file.svg" alt="" />
+	</li>
+</ul>
+```
```

Comment:
Quick test on this example.
Chrome gives the `li` a name of "New File". Firefox gives it a name of "·" (😖 it's the list marker, why default to that…) Edge gives it no name (and thus fails the rule).

I say we should at least have an Accessibility Support note on support for `title`. Maybe remove this example.

Reply:
Oh! I think I know why. The bullet is treated as text content, and so because there is text, the accessible name won't go to the title attribute. I'll update the example. That's a good catch!